server/circleci_test.go

```diff
@@ -37,3 +38,100 @@ func TestAreNotAllExpectedJobs(t *testing.T) {
 
 	assert.False(t, areAll)
 }
+
+func TestBlockPaths(t *testing.T) {
+	testcases := []struct {
+		name                 string
+		input                []*github.CommitFile
+		expectedError        bool
+		expectedMessageError string
+	}{
+		{
+			name: "A file is in the block list",
+			input: []*github.CommitFile{
+				{
+					Filename: github.String(".circleci/config.yml"),
+				},
+				{
+					Filename: github.String(".circleci/honk.test"),
+				},
+				{
+					Filename: github.String("build/validone-honk.go"),
+				},
+			},
+			expectedError:        true,
+			expectedMessageError: "The file `.circleci/config.yml` is in the blocklist and should not be modified from external contributors, please if you are part of the Mattermost Org submit this PR in the upstream.\n /cc @mattermost/core-security @mattermost/core-build-engineers",
```

Comment:
Comparing error strings is not very ideal. I think the objective here is to compare the filenames, and not the entire error message. If we have a custom error type as I mentioned above, we can solve this in a cleaner way.

Reply:
change that @agnivade